### DISCUSSION TITLE
Green Hydrogen Production (Project 204)

### DISCUSSION THREAD

**Discussion by robertogl** [mathworks/MATLAB-Simulink-Challenge-Project-Hub#35] [Collaborate]
Contribute to the discussion by asking and/or answering questions, commenting, or sharing your ideas for solutions to [project #204](https://github.com/mathworks/MathWorks-Excellence-in-Innovation/tree/main/projects/Green%20Hydrogen%20Production)

---
*:robot: An AI bot may reply to your comments with suggestions and guidance. It will not solve the project for you. Your comments may be processed by a third-party LLM. To direct a question to a specific person, tag them with @username.*

**Comment by JoshuaCNK** [comment]
Hello, my name is Joshua Curtis Kuffour, and I'm looking forward to working on this project! I'll try and provide updates as to the progress I've made on trying to use this package!

**Reply by gdudgeon** [reply]
Nice to have you on board Joshua. If you have any questions, please let me know.

**Comment by TheKachman** [comment]
Hi, I am Onyedikachi Ukawuba, I think this project is quite amazing!!
I look forward to working on it

**Reply by gdudgeon** [reply]
Nice to have you on board Onyedikachi. If you have any questions, please let me know.

**Comment by JoshuaCNK** [comment]
So I've created the three-phase system and it contains solar power. I'm not sure how exactly to implement energy storage. Is this where the battery part comes in or would there be a better form of energy to store it in?

**Reply by JoshuaCNK** [reply]
I've been thinking more about this and I'm stuck between using pumped hydro or an ultra/super capacitor. Given this is all heavily electrical, it would make sense to use an ultracapacitor and store for quick discharge, but it also seems like pumped hydro could be used as water is needed for electrolysis. If water is pumped up, the pumped water can be electrolyzed in the tank, and for longer term energy, the water stored can be sent downhill through a turbine to produce electricity.

**Reply by JoshuaCNK** [reply]
I think I'll try to implement both and see how things turn out.

**Reply by gdudgeon** [reply]
Hi Joshua. A main outcome of the project is to develop an electrolyzer model that can convert water to hydrogen. Take a look at the shipping example 'ssc_fuel_cell' for an example of how a fuel cell is implemented using Simscape language. You could use this model as inspiration to create an electrolyzer model.

**Comment by JoshuaCNK** [comment]
Alright, so as this project comes to a close (for me) I'll be uploading all things that have been used to help make progress. 
The moist air edition model utilizes the most accurate depiction to date of what the system should do. It is solely based on Simulink but assumes a current input that approximates what we would expect an electrolyzer to experience.
The "Controlled Water Entering Model" is also a somewhat accurate model. It doesn't have the current entering into anything (it is just a wire connected to itself) which is unrealistic but the behavior is simulated is about the same. 
The "Constant Irradiance Model" assumes no distribution of irradiance, just constant power entering in at all points in the day. I think it serves as a basis for checking the electrolyzer model found online. 
The "ee island" model essentially is the base point for all testing. It was here intuitions about the electrolyzing model were gained.
I'll also upload the many papers I consulted over the course of the project. Most of the equations used in the higher fidelity models can be found in the "Theoretical Model and Experimental Analysis of a high-pressure PEM water electrolyzer for hydrogen production"

**Comment by michaelfsb** [comment]
Hi, I am master's student at Federal University of Santa Catarina and interested in working and collaborating on this project.

**Comment by michaelfsb** [comment]
Hello.

I would like to discuss with you the reversible electrolysis process envisioned in the proposal. More specifically, I would like to know if the model developed in the project can be a discrete regenerative fuel cell, in which the electrolyzer and the fuel cell are separated devices; or an unitized fuel cell, where only an electrochemical device is used.
This is an important assumption for the mathematical modeling to be developed, as the equations (and the plant topology) will be different in both cases.

**Reply by michaelfsb** [reply]
Hello @gdudgeon. Can you help me with this question, please?

**Reply by gdudgeon** [reply]
Hi Michael. The process does not have to be reversible - you can model the electrolyzer as a separate device.

**Reply by michaelfsb** [reply]
Ok. Thanks.

**Comment by huntermanana** [comment]
Hello, my name is Manana Martin Hunter, and I'm looking forward to working on this project. 
I have begun on the project and would like to enquire about something, the image I have what is that component, what its use is, and how does one model it in Matlab?
![Capture](https://user-images.githubusercontent.com/124876929/228007397-fbd42c61-ba7b-43e5-b6cb-f965f3293b0a.PNG)

**Reply by huntermanana** [reply]
its part of the PEM Electrolysis System

**Reply by robertogl** [reply]
Hi @huntermanana,

That is just a [Simscape Bus block](https://www.mathworks.com/help/simscape/ref/simscapebus.html). In this case is used to bundle connections (H2O, O2) into a Simscape Bus line (out). As you can see, it is also used in the 'Recirculation' subsystem but this time to access the connections. 

I hope this helps.

**Reply by huntermanana** [reply]
thanks

**Comment by huntermanana** [comment]
hi,,
I hope you're well.
I have two inquiries,
1. What are the uses of these symbols in the attachments and what are they?
2. How do I model the Electron Membrane Assembly?

Thanks
![Capture 1](https://user-images.githubusercontent.com/124876929/230959791-33aca226-a172-4285-810b-fbec46b4328e.PNG)
![Capture 2](https://user-images.githubusercontent.com/124876929/230959809-70a978eb-adc7-427b-9c68-c711703b1d2c.PNG)

**Reply by robertogl** [reply]
Hi @huntermanana ,

That is a Separator Tank subsystem. As explained in the [documentation page](
https://www.mathworks.com/help/simscape/ug/pem-electrolysis-system.html#responsive_offcanvas) the Separator Tank models the balance of water and oxygen in the return flow before the oxygen is vented. The supply pump replenishes the system with fresh water.

I would suggest reading the documentation page thoroughly, along with any reference documents mentioned on the page.
The following references are the documents mentioned on the documentation page:

- Liso, Vincenzo, et al. "Modelling and experimental analysis of a polymer electrolyte membrane water electrolysis cell at different operating temperatures." Energies 11.12 (2018): 3273.

- Mo, Jingke, et al. "Thin liquid/gas diffusion layers for high-efficiency hydrogen production from water splitting." Applied Energy 177 (2016): 817-822. 

Cheers.

**Comment by satakshiatuta** [comment]
Hi,
My name is Satakshi Singh. I am an Electrical Engineering graduate student at University of Texas, Arlington. I am looking forward to working and completing this project along with my team member.

**Comment by TechWhiza** [comment]
Hello sir I am pursuing post graduation in chemical engineering, I am willing to work such a good project which have good future scope and some challanges
